### PR TITLE
feature/SOF-3779

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/MaterialsDesigner.js [hunk 1]
```diff
@@ -90,7 +92,14 @@ class MaterialsDesigner extends mix(React.Component).with(FullscreenComponentMix
...
                                     isConventionalCellShown={this.props.isConventionalCellShown}
-                                    onUpdate={this.props.onUpdate}
+                                    boundaryConditions={this.props.material.boundaryConditions}
+                                    onUpdate={(material) => {
+                                        // convert made material to MD material and re-set metadata
+                                        const config = material.toJSON();
+                                        const newMaterial = new Material(config);
+                                        newMaterial.metadata = this.props.material.metadata || {};
+                                        this.props.onUpdate(newMaterial);
+                                    }}
```
Comment (timurbazhirov) [inline src/MaterialsDesigner.js]: I suggest moving this to Material class - createFromMadeMaterial - function

File: src/components/3d_editor/advanced_geometry/BoundaryConditionsDialog.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import {ModalHeader, ModalBody, ModalFooter} from "react-bootstrap";
+
+import {ModalDialog} from "../../include/ModalDialog";
+
+const BOUNDARY_TYPES = {
+    "pbc": "Periodic Boundary Condition (pbc)",
+    "bc1": "Vacuum-slab-Vacuum (bc1)",
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/BoundaryConditionsDialog.js]: 1. Slab
2. Should this be in enums?

File: src/MaterialsDesigner.js (hunk 1 above)
Comment (mohammadidinani) [reply src/MaterialsDesigner.js]: Done.

File: src/components/3d_editor/advanced_geometry/BoundaryConditionsDialog.js (hunk 2 above)
Comment (mohammadidinani) [reply src/components/3d_editor/advanced_geometry/BoundaryConditionsDialog.js]: Done.

File: src/MaterialsDesigner.js (hunk 1 above)
Comment (timurbazhirov) [reply src/MaterialsDesigner.js]: I meant passing metadata as a second optional parameter to createFromMadeMaterial also, but it's up to you

**Pull request merged**